events/workers.go

```diff
@@ -0,0 +1,89 @@
+package events
+
+import (
+	"time"
+
+	diskqueue "github.com/nsqio/go-diskqueue"
+	"go.vocdoni.io/dvote/crypto/ethereum"
+	"go.vocdoni.io/dvote/log"
+	"go.vocdoni.io/dvote/types"
+	"go.vocdoni.io/dvote/vochain"
+	"go.vocdoni.io/proto/build/go/models"
+	"google.golang.org/protobuf/proto"
+)
+
+// eventCallback represents any callback function tied to an event source
+type eventCallback func(
+	app *vochain.BaseApplication,
+	signer *ethereum.SignKeys,
+	m proto.Message,
+) error
+
+// eventWorker is responsible for performing a unit of work given an event
+type eventWorker struct {
+	// storage is the disk queue
+	storage diskqueue.Interface
+	// signer
+	signer *ethereum.SignKeys
+	// vochainApp is a pointer to the Vochain BaseApplication allowing to call SendTx method
+	vochainApp *vochain.BaseApplication
+	// handlers maps each event to a callback using the event sourceType
+	handlers map[SourceType]eventCallback
+}
+
+// start starts the event worker.
+// The worker will constantly pull for new events
+// and will handle them one by one
+// Events can be handled sync or async and it is up
+// to the callback to define the behaviour.
+// i.e if we want OnProcessResults event to be processed
+// async the implementation of the handler must be wrapped with
+// an anonymous goroutine.
+func (ew *eventWorker) start() {
+	for {
+		// Receive a event.
+		event, err := ew.nextEvent()
+		if err != nil {
+			log.Warnf("cannot get next event: %s", err)
+			continue
+		}
+		if event == nil {
+			time.Sleep(time.Millisecond * types.EventWorkerSleepTime)
+			continue
+			// TODO: mvdan improve this
+		}
```

Comment:
this event nil check and sleep is now unnecessary, I think, as go-diskqueue will block until there's another event to receive.